client/src/main/java/com/tigrisdata/db/client/model/UpdateFields.java

```diff
@@ -0,0 +1,99 @@
+package com.tigrisdata.db.client.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.tigrisdata.db.client.utils.Utilities;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public final class UpdateFields {
+  private final Map<String, Object> internalMap;
+  private static final UpdateFields EMPTY = new UpdateFields(Collections.emptyMap());
+  private static final String SET_OPERATOR = "$set";
+
+  private UpdateFields(UpdateFieldsBuilder builder) {
+    this.internalMap = builder.map;
+  }
+
+  private UpdateFields(Map<String, Object> internalMap) {
+    this.internalMap = internalMap;
+  }
+
+  public static UpdateFields empty() {
+    return EMPTY;
+  }
+
+  public String toJSON() throws JsonProcessingException {
+    return Utilities.OBJECT_MAPPER.writeValueAsString(internalMap);
+  }
+
+  public static UpdateFieldsBuilder newBuilder() {
+    return new UpdateFieldsBuilder();
+  }
+
+  public static class UpdateFieldsBuilder {
+    private final Map<String, Object> map;
+
+    private UpdateFieldsBuilder() {
+      this.map = new LinkedHashMap<>();
+    }
+
+    public UpdateFieldsBuilder set(SetFields setFields) {
+      this.map.put(SET_OPERATOR, setFields.getInternalMap());
+      return this;
+    }
+
+    public UpdateFields build() {
+      return new UpdateFields(this);
+    }
+  }
+
+  public static class SetFields {
+    private final Map<String, Object> internalMap;
+
+    private SetFields(SetFieldsBuilder builder) {
+      this.internalMap = builder.internalMap;
+    }
+
+    private Map<String, Object> getInternalMap() {
+      return Collections.unmodifiableMap(internalMap);
+    }
+
+    public static SetFieldsBuilder newBuilder() {
+      return new SetFieldsBuilder();
+    }
+  }
+
+  public static class SetFieldsBuilder {
+    private final Map<String, Object> internalMap;
+
+    private SetFieldsBuilder() {
+      this.internalMap = new LinkedHashMap<>();
+    }
+
+    public SetFieldsBuilder set(String fieldName, int newValue) {
```

Comment:
I am assuming you are planning to add more setters for bytes etc?

Reply:
discussed offline and added `byte[]`